entry/src/block_component.rs

```diff
@@ -415,6 +438,23 @@ impl BlockComponent {
         matches!(self, Self::BlockMarker(_))
     }
 
+    /// Fuses another BlockComponent into this one if both are entry batches.
+    ///
+    /// Returns None if fusion occurred, Some(other) if no fusion was possible.
+    pub fn maybe_fuse(&mut self, other: Self) -> Option<Self> {
```

Comment:
```suggestion
    pub fn try_fuse(&mut self, other: Self) -> Option<Self> {
```

nit: I think this is a more intuitive name for this function.